### PR TITLE
Updates for no longer logging to files in production

### PR DESCRIPTION
**Pull request by agrare**
TODO
- [x] Update references to `evm.log` etc... to journalctl and oc logs
- [x] Add examples of using `journalctl` ref https://github.com/ManageIQ/manageiq/pull/21177#issuecomment-923262365
- [x] Add examples of using `oc logs`

https://github.com/ManageIQ/manageiq/pull/21177

### CODE REVIEW

File: managing_log_files/_topics/appliance_journald_logging.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Logs for a specific unit: `journalctl -f -u manageiq-generic@51113329-8942-4a4c-98d6-1e532656c569.service`
+Logs for a group of workers: `journalctl -f -u manageiq-generic.slice`
+Logs matching a pattern: `journalctl -g '(Started|Stopping) EVM server''` (accepts perl compatible regular expressions)
+Logs for a specific identifier: `journalctl -f -t evm` (equivalent of `tail -f log/evm.log`)
+
+Rotate the current journal to an archive: `journalctl --rotate`
+Clear out all archived journals: `journalctl --vacuum-time=1s`
+Clear out journal archives over a day old: `journalctl --vacuum-time=1d`
+
+Almost all of these can be combined together where they don't conflict (e.g. -f can't be used with -r)
```
Comment (agrare) [inline managing_log_files/_topics/appliance_journald_logging.md:27]: NOTE very WIP just wanted to drop these examples in here for now

File: managing_log_files/index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+---
+
+# Managing Log Files
+
+{% include_relative _topics/appliance_journald_logging.md %}
```
Comment (agrare) [inline managing_log_files/index.md:6]: TODO add container examples

File: managing_log_files/index.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+---
+
+# Managing Log Files
```
Comment (agrare) [inline managing_log_files/index.md:4]: NOTE I didn't see any good place to drop in examples of working with log files so thought a new section under "Administration" would work, open to other ideas and/or names for this section

Comment (agrare) [reply managing_log_files/index.md:4]: Yeah I like troubleshooting, I had been thinking of "Debugging" but that sounded too developer focused for these docs but troubleshooting sounds good.

File: troubleshooting/_topics/logging.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+{{ site.data.product.title_short }} has extensive logging to assist in troubleshooting issues or just seeing what is being worked on.
+
+On appliances logging is managed using the built-in systemd-journal system.  Journald uses database-like binary files to store log lines along with extensive metadata.
+
+There is a `journalctl` command-line utility which can be used to query the journal.
+
+If you are used to using `tail -f /var/www/miq/vmdb/log/evm.log` then you can simply switch to `journalctl -f -t evm`.  Logging which would have previously gone to one of the other log files like `automation.log` can now be viewed using the syslog_identifier e.g. `journalctl -f -t automation`.
+
+But `journalctl` allows you to do much more like view logs for a specific unit (aka worker), for a time range, and for a group of workers (e.g. every generic worker).
```
Comment (Fryguy) [inline troubleshooting/_topics/logging.md:11]:
```suggestion
`journalctl` allows you to do much more like view logs for a specific unit (aka worker), for a time range, and for a group of workers (e.g. every generic worker).
```

File: troubleshooting/_topics/logging.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+View all logs from a time range: `journalctl --since "1 hour ago"`
+View all logs with extra metadata: `journalctl --output verbose`
+
+Tail all logs: `journalctl -f`
+Tail all warnings: `journalctl -f -p warning`
+Tail all errors: `journalctl -f -p err`
+Tail all warnings and errors: `journalctl -f -p err -p warning`
+
+Logs for a specific unit: `journalctl -f -u manageiq-generic@51113329-8942-4a4c-98d6-1e532656c569.service`
+Logs for a group of workers: `journalctl -f -u manageiq-generic.slice`
```
[the remark this answers is not in the document]
Comment (agrare) [reply troubleshooting/_topics/logging.md:28]: Service names are not productized

File: troubleshooting/_topics/logging.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+### Logging
+
+{{ site.data.product.title_short }} has extensive logging to assist in troubleshooting issues or just seeing what is being worked on.
+
+On appliances logging is managed using the built-in systemd-journal system.  Journald uses database-like binary files to store log lines along with extensive metadata.
+
+There is a `journalctl` command-line utility which can be used to query the journal.
```
Comment (Fryguy) [inline troubleshooting/_topics/logging.md:7]: We might want to link out to external documentation for journalctl.  If not here, then maybe below, with something like "For more detailed information see the [man page for journalctl](https://man7.org/linux/man-pages/man1/journalctl.1.html)"

File: troubleshooting/_topics/logging.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+Tail all logs: `journalctl -f`
+Tail all warnings: `journalctl -f -p warning`
+Tail all errors: `journalctl -f -p err`
+Tail all warnings and errors: `journalctl -f -p err -p warning`
+
+Logs for a specific unit: `journalctl -f -u manageiq-generic@51113329-8942-4a4c-98d6-1e532656c569.service`
+Logs for a group of workers: `journalctl -f -u manageiq-generic.slice`
+Logs matching a pattern: `journalctl -g '(Started|Stopping) EVM server''` (accepts perl compatible regular expressions)
+Logs for a specific identifier: `journalctl -f -t evm` (equivalent of `tail -f log/evm.log`)
```
Comment (Fryguy) [inline troubleshooting/_topics/logging.md:30]: Similar comment here about the "old way"

File: troubleshooting/_topics/logging.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+Export to a file: `journalctl --output export > export.log`
+Rotate the current journal to an archive: `journalctl --rotate`
+Clear out all archived journals: `journalctl --vacuum-time=1s`
+Clear out journal archives over a day old: `journalctl --vacuum-time=1d`
+
+Almost all of these can be combined together where they don't conflict (e.g. -f can't be used with -r)
```
Comment (Fryguy) [inline troubleshooting/_topics/logging.md:37]:
```suggestion
Almost all of these can be combined together where they don't conflict (e.g. `-f` can't be used with `-r`)
```

File: troubleshooting/_topics/podified_logging.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+### Podified Logging
+
+When running in podified all logs are output to the container's stdout which is picked up by Kubernetes/OpenShift and made available via `kubectl logs`/`oc logs`.
```
Comment (agrare) [inline troubleshooting/_topics/podified_logging.md:3]: TODO I'd like to add some of the examples that @jaywcarman showed where you can pass a label to `oc logs` as well as @Fryguy's json log tool

Comment (Fryguy) [reply troubleshooting/_topics/podified_logging.md:3]: That json tool still isn't merged: https://github.com/ManageIQ/manageiq/pull/21942

Comment (agrare) [reply troubleshooting/_topics/podified_logging.md:3]: Okay I've added an example of using the label selector, will update with the json utility once it is merged

**Comment by agrare** [comment]
Podified examples added, taking out of WIP

**Review by Fryguy — APPROVED** [review]

**Comment by Fryguy** [comment]
Not yet no.

**Comment by Fryguy** [comment]
Oh wait, this is docs - yeah we can merge this.  @agrare Please do a followup to add this to the website sidebar menu.

**Comment by Fryguy** [comment]
Actually, the menu for the docs is in this repo (I always get confused cause the menu for _API_ docs is in the other repo).  It's here: https://github.com/ManageIQ/manageiq-documentation/blob/master/_data/site_menu.yml

**Comment by agrare** [comment]
I thought the sidebar was in the manageiq.org repo also, added a new Troubleshooting section

**Pull request merged**